### PR TITLE
External config

### LINKED ISSUE #43: Centralize our configuration
To follow DYR, most (if not all) error and status messages are on `backend/src/configs.clj` (maybe it should be nested somewhere else?)

There are still some of those hardcoded on `frontend/test/unit/Game.spec.js` (lines 16, 26 and 29).
(I couldn't find any other case where that happens)

Those should make reference to what's on `backend/src/configs.clj`.

### PR DESCRIPTION
**Pull request by kenan-rhoton**
Closes #43.

Still need to do readability changes all around.

### CODE REVIEW

File: frontend/lobby/waiting/waiting.js [hunk 1]
```diff
@@ -5,6 +5,17 @@ const gameID = params.get("gameID");
 const playerID = params.get("playerID");
 const gameURL = `/game/?gameID=${gameID}&playerID=${playerID}`;
 
+var config = {}
+
```
Comment (kenan-rhoton) [inline frontend/lobby/waiting/waiting.js]: This is a super ugly section, I'm working on it.

File: configs/config.yml [hunk 2]
```diff
@@ -0,0 +1,7 @@
+# Anything in this file should be considered 100% public as it is served to frontend
+messages:
+    out-of-turn: "Our of turn play"
```
Comment (Masclins) [inline configs/config.yml]: *Out*

File: frontend/lobby/waiting/waiting.js [hunk 3]
```diff
@@ -13,7 +24,7 @@ function checkForOpponent() {
     req.onload = function () {
         const status = req.response["status"];
         
-        if (status !== "Waiting for an opponent") {
+        if (status !== config.messages["no-opp"]) {
```
Comment (Masclins) [inline frontend/lobby/waiting/waiting.js]: Funny how many things had to change only for this line. 
Thanks!

File: e2e/test.js [hunk 4]
```diff
@@ -16,7 +21,7 @@ test('Create a game', async testCase => {
         .expect(gamePage.gameStatus.exists).notOk()
         .click(mainPage.createGame)
         .expect(gamePage.joinLink.exists).ok()
-        .expect(gamePage.gameStatus.innerText).eql("Waiting for opponent")
+        .expect(gamePage.gameStatus.innerText).eql(config.messages["wait-for-opponent"])
```
Comment (Masclins) [inline e2e/test.js]: it should be `config.messages["no-opp"]`

File: frontend/lobby/waiting/index.html [hunk 5]
```diff
@@ -16,14 +16,15 @@ <h1 class="display-1">
...
                 <div id="game-status" class="col-12">
-                    Waiting for opponent
                 </div>
             </div>
             <template id="joinTemplate">
                 <div id="join-link" class="col-12">
                 </div>
             </template>
         </div>
+        <script src="/config/js-yaml.min.js"></script>
```
Comment (Masclins) [inline frontend/lobby/waiting/index.html]: Can't find this file.

Comment (kenan-rhoton) [reply frontend/lobby/waiting/index.html]: it's in `frontend/config/js-yaml.min.js`.

You have to be on this branch to see it.

Comment (Masclins) [reply frontend/lobby/waiting/index.html]: I'd bet it's not a file you handwritted...

Comment (kenan-rhoton) [reply frontend/lobby/waiting/index.html]: it's a library 😂

**Pull request merged**